cmd/govim/main_test.go

```diff
@@ -50,75 +52,128 @@ func TestScripts(t *testing.T) {
+					e.Vars = append(e.Vars,
+						"TMPDIR="+tmp,
+						"HOME="+home,
+						"PLUGIN_PATH="+govimPath,
+						"CURRENT_GOPATH="+strings.TrimSpace(runCmd(t, "go", "env", "GOPATH")),
+					)
+					testPluginPath := filepath.Join(e.WorkDir, "home", ".vim", "pack", "plugins", "start", "govim")
 
-				errLog := new(testdriver.LockingBuffer)
-				outputs := []io.Writer{
-					errLog,
-				}
-				e.Values[testdriver.KeyErrLog] = errLog
-				if os.Getenv(testsetup.EnvTestscriptStderr) == "true" {
-					outputs = append(outputs, os.Stderr)
-				}
+					errLog := new(testdriver.LockingBuffer)
+					outputs := []io.Writer{
+						errLog,
+					}
+					e.Values[testdriver.KeyErrLog] = errLog
+					if os.Getenv(testsetup.EnvTestscriptStderr) == "true" {
+						outputs = append(outputs, os.Stderr)
+					}
 
-				if *fDebugLog {
-					tf, err := ioutil.TempFile("", "govim_test_script_govim_log*")
-					if err != nil {
-						t.Fatalf("failed to create govim log file: %v", err)
+					if *fDebugLog {
+						tf, err := ioutil.TempFile("", "govim_test_script_govim_log*")
+						if err != nil {
+							t.Fatalf("failed to create govim log file: %v", err)
+						}
+						outputs = append(outputs, tf)
+						t.Logf("logging %v to %v\n", filepath.Base(e.WorkDir), tf.Name())
 					}
-					outputs = append(outputs, tf)
-					t.Logf("logging %v to %v\n", filepath.Base(e.WorkDir), tf.Name())
-				}
 
-				d := newplugin(string(goplspath))
+					var defaults *config.Config
 
-				config := &testdriver.Config{
-					Name:           filepath.Base(e.WorkDir),
-					GovimPath:      govimPath,
-					Log:            io.MultiWriter(outputs...),
-					TestHomePath:   home,
-					TestPluginPath: testPluginPath,
-					Env:            e,
-					Plugin:         d,
-				}
-				td, err := testdriver.NewTestDriver(config)
-				if err != nil {
-					t.Fatalf("failed to create new driver: %v", err)
-				}
-				if err := td.Run(); err != nil {
-					t.Fatalf("failed to run TestDriver: %v", err)
-				}
-				waitLock.Lock()
-				waitList = append(waitList, td.Wait)
-				waitLock.Unlock()
-				e.Defer(func() {
-					td.Close()
-				})
-				return nil
-			},
+					// check whether we have a default config to apply
+					configPath := filepath.Join("testdata", entry.Name(), "default_config.json")
+					if fi, err := os.Stat(configPath); err == nil {
+						if fi.Mode().IsRegular() {
+							config, err := ioutil.ReadFile(configPath)
+							if err != nil {
+								t.Fatalf("failed to read %v: %v", configPath, err)
+							}
+							if err := json.Unmarshal(config, &defaults); err != nil {
+								t.Fatalf("failed to unmarshal JSON config in %v: %v\n%s", configPath, err, config)
+							}
+							// Now verify that we haven't supplied superfluous JSON
+							var i interface{}
+							if err := json.Unmarshal(config, &i); err != nil {
+								t.Fatalf("failed to re-parse JSON config in %v: %v\n%s", configPath, err, config)
+							}
+							var defaultsCheck []byte
+							{
```

Comment:
What is the purpose of this additional scope?

Reply:
Nothing more than an artefact of a refactoring. Fixed.